cli/resolver.rs

```diff
@@ -371,6 +398,52 @@ impl Resolver for CliGraphResolver {
   }
 }
 
+fn sloppy_imports_resolve(
+  resolver: &UnstableSloppyImportsResolver,
+  specifier: ModuleSpecifier,
+  referrer_range: &deno_graph::Range,
+) -> ModuleSpecifier {
+  let resolution = resolver.resolve(&specifier);
+  let message = match &resolution {
+    UnstableSloppyImportsResolution::JsToTs(to_specifier) => {
+      let from_media_type = MediaType::from_specifier(&specifier);
+      let to_media_type = MediaType::from_specifier(to_specifier);
+      format!(
+        "(update {} extension to {})",
+        from_media_type.as_ts_extension(),
+        to_media_type.as_ts_extension()
+      )
+    }
+    UnstableSloppyImportsResolution::NoExtension(to_specifier) => {
+      let to_media_type = MediaType::from_specifier(to_specifier);
+      format!("(add {} extension)", to_media_type.as_ts_extension())
+    }
+    UnstableSloppyImportsResolution::Directory(to_specifier) => {
+      let file_name = to_specifier
+        .path()
+        .rsplit_once('/')
+        .map(|(_, file_name)| file_name)
+        .unwrap_or(to_specifier.path());
+      format!("(specify path to {} file in directory instead)", file_name)
+    }
+    UnstableSloppyImportsResolution::None(_) => return specifier,
+  };
+  // show a warning when this happens in order to drive
+  // the user towards correcting these specifiers
+  log::warn!(
+    "{} Sloppy import resolution {}\n    at {}",
+    crate::colors::yellow("Warning"),
+    message,
```

Comment:
Maybe we could do something like: 
`Warning Sloppy import resolution (hint: add .mjs extension)`; where `hint` part is in gray?